service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -779,15 +781,18 @@ private JsonNode getOkapiHoldings(String instanceId) {
         Map<String, Map<String, String>> okapiItems = new HashMap<String, Map<String, String>>();
         if (itemsResponse.isObject()) {
             itemsResponse.at("/items").forEach(i -> {
+                JsonNode itemLocationNode = getLocation(i.at("/effectiveLocationId").asText());
                 JsonNode loanType = getOkapiLoanType(i.at("/permanentLoanTypeId").asText());
                 Map<String, String> itemData = new HashMap<String, String>();
                 itemData.put("hrid", i.at("/hrid").asText());
                 itemData.put("barcode", i.at("/barcode").asText());
-                itemData.put("locationCode", getLocation(i.at("/effectiveLocationId").asText()).at("/code").asText());
+                itemData.put("locationCode", itemLocationNode.at("/code").asText());
+                itemData.put("location", itemLocationNode.at("/name").asText());
                 itemData.put("enumeration", i.at("/enumeration").asText());
                 itemData.put("chron", i.at("/chronology").asText());
                 itemData.put("status", i.at("/status/name").asText());
                 itemData.put("typeDesc", loanType.at("/name").asText());
+                itemData.put("callNumber", i.at("/effectiveCallNumberComponents").at("/callNumber").asText());
```

Comment:
This can be simplified.

```
getText(i, "/effectiveCallNumberComponents/callNumber");
```

Reply:
Updated to use '/effectiveCallNumberComponents/callNumber'